src/compiler/checker.ts

```diff
@@ -12030,7 +12030,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     /**
      * The base constructor of a class can resolve to
      * * undefinedType if the class has no extends clause,
-     * * unknownType if an error occurred during resolution of the extends expression,

```

Comment:
Based on my analysis and the existing test suite - this can't actually be returned. Perhaps `checkExpression` might return `unknownType` but that would fail the `isConstructorType` check and thus it would lead to returning `errorType`

Reply:
Should this say `errorType`, then?

Reply:
Ye, not sure why I removed it instead of adjusting it. I just brought it back 😉